sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala

```diff
@@ -165,8 +181,22 @@ case class GpuGetMapValue(child: Expression, key: Expression)
 
   override def prettyName: String = "getMapValue"
 
-  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector =
-    lhs.getBase.getMapValue(rhs.getBase)
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
+    if (failOnError){
+      withResource(lhs.getBase.getMapKeyExistence(rhs.getBase)) { keyExistenceColumn =>
+        withResource(keyExistenceColumn.all) { exist =>
+          if (exist.getBoolean) {
+            return lhs.getBase.getMapValue(rhs.getBase)
+          } else {
+            throw new NoSuchElementException(s"Key: ${rhs.getValue.asInstanceOf[String]} does " +
```

Comment:
getValue is supposed to return a UTF8String.

Reply:
Thanks, forgot to add integration test for this case so I didn't catch it.